packages/video_player_videohole/lib/video_player.dart

```diff
@@ -0,0 +1,1223 @@
+          controller.play();
+        }
+      },
+      onTapDown: (TapDownDetails details) {
+        if (!controller.value.isInitialized) {
+          return;
+        }
+        seekToRelativePosition(details.globalPosition);
+      },
+    );
+  }
+}
+
+/// Displays the play/buffering status of the video controlled by [controller].
+///
+/// If [allowScrubbing] is true, this widget will detect taps and drags and
+/// seek the video accordingly.
+///
+/// [padding] allows to specify some extra padding around the progress indicator
+/// that will also detect the gestures.
+class VideoProgressIndicator extends StatefulWidget {
+  /// Construct an instance that displays the play/buffering status of the video
+  /// controlled by [controller].
+  ///
+  /// Defaults will be used for everything except [controller] if they're not
+  /// provided. [allowScrubbing] defaults to false, and [padding] will default
+  /// to `top: 5.0`.
+  const VideoProgressIndicator(
+    this.controller, {
+    this.colors = const VideoProgressColors(),
+    required this.allowScrubbing,
+    this.padding = const EdgeInsets.only(top: 5.0),
+  });
+
+  /// The [VideoPlayerController] that actually associates a video with this
+  /// widget.
+  final VideoPlayerController controller;
+
+  /// The default colors used throughout the indicator.
+  ///
+  /// See [VideoProgressColors] for default values.
+  final VideoProgressColors colors;
+
+  /// When true, the widget will detect touch input and try to seek the video
+  /// accordingly. The widget ignores such input when false.
+  ///
+  /// Defaults to false.
+  final bool allowScrubbing;
+
+  /// This allows for visual padding around the progress indicator that can
+  /// still detect gestures via [allowScrubbing].
+  ///
+  /// Defaults to `top: 5.0`.
+  final EdgeInsets padding;
+
+  @override
+  _VideoProgressIndicatorState createState() => _VideoProgressIndicatorState();
+}
+
+class _VideoProgressIndicatorState extends State<VideoProgressIndicator> {
+  _VideoProgressIndicatorState() {
+    listener = () {
+      if (!mounted) {
+        return;
+      }
+      setState(() {});
+    };
+  }
+
+  late VoidCallback listener;
+
+  VideoPlayerController get controller => widget.controller;
+
+  VideoProgressColors get colors => widget.colors;
+
+  @override
+  void initState() {
+    super.initState();
+    controller.addListener(listener);
+  }
+
+  @override
+  void deactivate() {
+    controller.removeListener(listener);
+    super.deactivate();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    Widget progressIndicator;
+    if (controller.value.isInitialized) {
+      final int duration = controller.value.duration.inMilliseconds;
+      final int position = controller.value.position.inMilliseconds;
+
+      progressIndicator = Stack(
+        fit: StackFit.passthrough,
+        children: <Widget>[
+          LinearProgressIndicator(
```

Comment:
The `controller.value.buffered` value is being set by `eventListener` but is it ever used? The buffering status indicator widget which was present in the original code is gone.

Reply:
`buffered ` is used in [method_channel_video_player.dart](https://github.com/flutter-tizen/plugins/blob/69660ad54ac718e9294a67c8e4fd9c7a1584cb91/packages/video_player_videohole/lib/src/method_channel_video_player.dart#L123)
It is used to listen to stream bufferupdate.

Reply:
Did you understand the second sentence in my comment? That's why I referenced this specific line of code (1113). My question is: was there any reason that you removed the `LinearProgressIndicator` from the `VideoProgressIndicator` widget tree?

Reply:
Okay, i understand now. In video_hole we don't use `controller.value.buffered`, because the buffer mechanism is different from video_player. 
In video_player_tizen, buffer is based on buffer space, not percent(in video_player), `LinearProgressIndicator` which display buffer progress  is unnecessary.
So we removed it, but add `LinearProgressIndicator` also be no problem.